internal/action/workspace.go

```diff
@@ -290,3 +291,30 @@ func FindWorkspace(workspaces []*Workspace, target string) *Workspace {
 
 	return nil
 }
+
+// ParseWorkspaceYAML takes list of YAML encoded workspaces and the generic workspace name, and returns a list of Workspace objects
+func ParseWorkspaceYAML(input string, name string) ([]*Workspace, error) {
```

Comment:
ParseWorkspaces..., sorry 😛

This function also seems to be doing a bit too much. It says it parses YAML but it's introducing workspace semantics too. 

Keep YAML parsing and business logic separate. For example, `len(workspaces) == 0` is better than `workspaceString == ""`. It's not just more readable—the latter introduces subtle bugs. It fails to handle `workspaces: []` correctly, among others.

Reply:
gp regarding the empty list!